spec/repositories/section_repository_spec.rb

```diff
@@ -70,23 +70,23 @@ def build_published_edition(version: 1)
     end
 
     it "populates the section with all editions for that section id" do
-      section_repository[section_id]
+      section_repository.fetch(section_id)
 
       expect(Section).to have_received(:build)
         .with(manual: manual, id: section_id, editions: editions)
     end
 
     it "returns the section" do
-      expect(section_repository[section_id]).to eq(section)
+      expect(section_repository.fetch(section_id)).to eq(section)
     end
 
     context "when there are no editions" do
       before do
         allow(editions_proxy).to receive(:to_a).and_return([])
       end
 
-      it "returns nil" do
-        expect(section_repository[section_id]).to be_nil
+      it "raises a key error" do
+        expect { section_repository.fetch(section_id) }.to raise_error(KeyError)
```

Comment:
I've added the following to the commit note: 

> I'm happy to accept this change in behaviour because we were never calling `#[]` directly: we only ever used `#fetch` which invoked `#[]`.